travis-scripts/valgrind.sh

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env bash
+set -e
+
+if [ -d /var/cfengine ]; then
+    rm -rf /var/cfengine
+fi
+
+# Test assumes we start in core / masterfiles directory
+cd ../
+
+if [ ! -d core ]; then
+    echo "Cloning core (master)"
+    git clone --recursive https://github.com/cfengine/core.git
+fi
+
+if [ ! -d masterfiles ]; then
+    echo "Cloning masterfiles (master)"
+    git clone --recursive https://github.com/cfengine/masterfiles.git
```

Comment:
cloning whole repo, not shallow?

Reply:
`--recursive` includes submodules. This part is only included because the script also runs in core travis.